jib-core/src/main/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.java

```diff
@@ -213,18 +212,55 @@ public CredentialRetriever dockerConfig(Path dockerConfigFile) {
         new DockerConfigCredentialRetriever(imageReference.getRegistry(), dockerConfigFile));
   }
 
+  /**
+   * Creates a new {@link CredentialRetriever} that tries to retrieve credentials from <a
+   * href="https://cloud.google.com/docs/authentication/production">Google Application Default
+   * Credentials.</a>
+   *
+   * @return a new {@link CredentialRetriever}
+   * @see <a
+   *     href="https://cloud.google.com/docs/authentication/production">https://cloud.google.com/docs/authentication/production</a>
+   */
+  public CredentialRetriever googleApplicationDefaultCredentials() {
+    return () -> {
+      try {
+        if (imageReference.getRegistry().endsWith("gcr.io")) {
+          GoogleCredentials googleCredentials = googleCredentialsProvider.get();
+          logger.accept(LogEvent.info("Google ADC found"));
+          if (googleCredentials.createScopedRequired()) { // Not scoped if service account.
+            // The short-lived OAuth2 access token generated from the service account will have
+            // one-hour expiry (as of Aug 2019). Instead of using an access token, It is technically
+            // possible to use the service account private key to auth with GCR, but it does not
+            // worth writing complex code to achieve that.
+            logger.accept(LogEvent.info("ADC is a service account. Set GCS read-write scope"));
+            List<String> scope = Collections.singletonList(OAUTH_SCOPE_STORAGE_READ_WRITE);
+            googleCredentials = googleCredentials.createScoped(scope);
+          }
+          googleCredentials.refresh();
```

Comment:
Would it make sense to use `refreshIfExpired()`?  It checks if the access token expiry is soonish (looks like 5 minutes).